content/docs/base-components/ReactiveComponent.md

```diff
@@ -0,0 +1,83 @@
+---
+id: reactivecomponent
+title: "Reactive Component"
+layout: docs
+sectionid: reactivecomponent
+permalink: base-components/reactivecomponent.html
+prev: selectedfilters.html
+prevTitle: "SelectedFilters"
+next: /list-components/singlelist.html
+nextTitle: "List Components: SingleList"
+redirect_from:
+    - 'basic-components/reactivecomponent.html'
+    - 'base-components/reactivecomponent'
+    - 'reactivecomponent'
+---
+
+
+With `ReactiveComponent`, you can convert any React Component into a Reactivesearch component i.e. your react component will be able to talk to other Reactivesearch components and connect with your elasticsearch cluster seamlessly. `Note` This page only cover `ReactiveComponent` with `customQuery`. Read more [here](/advanced/reactivecomponentindepth.html).
+
+> How does this work?
+>
+> `ReactiveComponent` is a wrapper component that allows you to connect custom component(s) (passed as children) with the Reactivesearch ecosystem.
+
+
+### Usage
+
+Let's suppose - we are building an e-commerce store for cars which displays a list of cars of a particular brand on their separate page as `example.com/cars/nissan`. Now, we want all the filters on that page (like pricing, type of car, model, year, etc) to only show the data relevant to the given brand (i.e. `nissan`). In this case, `ReactiveComponent` can be used with `customQuery` to achieve the desired behavior easily.
+
+We can then use the given ReactiveComponent to be watched by all the filters (via `react` prop) to avail the desired brand based filtering for all the filters.
+
+Check demo [here](https://codesandbox.io/s/3ylrrr0r5q).
+
+
+```js
+ <ReactiveComponent
+    componentId="CarSensor"
+    customQuery={props => ({
```

Comment:
Use `defaultQuery` here instead. And use `onAllData` to show the query execution. We can also mix the `render` prop here to render a simple JSX UI label showing the hits found.